bindings/wasm/src/tangle/client_config.rs

```diff
@@ -0,0 +1,345 @@
+        .wasm_result()?;
+    }
+    if let Some(NodeAuth {
+      url,
+      jwt,
+      username,
+      password,
+    }) = options.primary_pow_node
+    {
+      builder = builder
+        .primary_pow_node(&url, jwt, username.as_deref().zip(password.as_deref()))
+        .wasm_result()?;
+    }
+    if let Some(node_auth) = options.node_auth {
+      for NodeAuth {
+        url,
+        jwt,
+        username,
+        password,
+      } in node_auth
+      {
+        builder = builder
+          .node_auth(&url, jwt, username.as_deref().zip(password.as_deref()))
+          .wasm_result()?;
+      }
+    }
+    if let Some(node_sync_interval) = options.node_sync_interval {
+      builder = builder.node_sync_interval(Duration::from_secs(u64::from(node_sync_interval)));
+    }
+    if let Some(node_sync_disabled) = options.node_sync_disabled {
+      if node_sync_disabled {
+        builder = builder.node_sync_disabled();
+      }
+    }
+    if let Some(quorum) = options.quorum {
+      builder = builder.quorum(quorum);
+    }
+    if let Some(quorum_size) = options.quorum_size {
+      builder = builder.quorum_size(quorum_size);
+    }
+    if let Some(quorum_threshold) = options.quorum_threshold {
+      builder = builder.quorum_threshold(quorum_threshold);
+    }
+    if let Some(local_pow) = options.local_pow {
+      builder = builder.local_pow(local_pow);
+    }
+    if let Some(fallback_to_local_pow) = options.fallback_to_local_pow {
+      builder = builder.fallback_to_local_pow(fallback_to_local_pow);
+    }
+    if let Some(tips_interval) = options.tips_interval {
+      builder = builder.tips_interval(u64::from(tips_interval));
+    }
+    if let Some(request_timeout) = options.request_timeout {
+      builder = builder.request_timeout(Duration::from_secs(u64::from(request_timeout)));
+    }
+
+    Ok(builder)
+  }
+}
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "IClientConfig")]
+  pub type IClientConfig;
+}
+
+/// Helper-struct for deserializing [`INodeAuth`].
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+struct NodeAuth {
+  url: String,
+  jwt: Option<String>,
+  username: Option<String>,
+  password: Option<String>,
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const I_NODE_AUTH: &'static str = r#"
+/** IOTA node details with optional authentication. */
+interface INodeAuth {
+    readonly url: string;
+    readonly jwt?: string;
+    readonly username?: string;
+    readonly password?: string;
+}"#;
+
+/// Helper-struct for deserializing [`IConfigOptions`].
+#[derive(Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct ConfigOptions {
+  network: Option<Network>,
+  encoding: Option<WasmDIDMessageEncoding>,
+  nodes: Option<Vec<String>>,
+  primary_node: Option<NodeAuth>,
+  primary_pow_node: Option<NodeAuth>,
+  node_auth: Option<Vec<NodeAuth>>,
+  node_sync_interval: Option<u32>,
+  node_sync_disabled: Option<bool>,
+  node_pool_urls: Option<Vec<String>>,
```

Comment:
Thanks for the catch. I couldn't figure out a way to handle it gracefully since it's the only option which requires an `async` call, so I decided to remove it for now.